browser/src/slideshow/transition3d/PermTextureTransition.ts

```diff
@@ -22,35 +22,32 @@ class PermTextureTransition extends SimpleTransition {
 	}
 
 	public createPermTexture(): WebGLTexture {
-		const permTexture = this.gl.createTexture();
-		this.gl.bindTexture(this.gl.TEXTURE_2D, permTexture);
+		const gl = this.gl;
+		const permTex = gl.createTexture();
+		gl.bindTexture(gl.TEXTURE_2D, permTex);
+
+		gl.texStorage2D(gl.TEXTURE_2D, 1, gl.RGBA8, 256, 256);
+
 		const data = this.generatePermutationData();
-		this.gl.texImage2D(
-			this.gl.TEXTURE_2D,
+		gl.texSubImage2D(
+			gl.TEXTURE_2D,
+			0,
+			0,
 			0,
-			this.gl.RGBA,
 			256,
 			256,
-			0,
-			this.gl.RGBA,
-			this.gl.UNSIGNED_BYTE,
+			gl.RGBA,
+			gl.UNSIGNED_BYTE,
```

Comment:
Does this not just have the same effect as changing the internal format to RGB? It's unlikely these days, but changing from texImage2D to texSubImage2D may have unwanted side-effects too, I'd be keen to hear the motivation behind this.

Reply:
Whoops, I misread RGBA8 as RGB8 - in that case, what effect does this have at all?

Reply:
By using `texStorage2D` followed by `texSubImage2D`, we allocate immutable GPU storage up front and then upload our pixel data into it, which can avoid hidden reallocations or format conversions. For a single 256×256 RGBA8 texture, however, the performance benefit is negligible, so unless we’ve measured a real issue here, it isn’t worth the extra complexity.

Reply:
I can buy that :)